src/kernel/tcap.c

```diff
@@ -215,6 +218,30 @@ tcap_merge(struct tcap *dst, struct tcap *rm)
 	return 0;
 }
 
+int
+tcap_wakeup(struct tcap *tc, tcap_prio_t prio, struct thread *thd, struct cos_cpu_local_info *cli)
+{
```

Comment:
I think that we also need the `budget`.  As the code stands, we're setting the next timer interrupt to fire corresponding to the budget of the associated tcap.  But we want the scheduler to be able to instead control when the next timer interrupt fires, so long as it is less than the tcap's budget.

This is, in the current design, the difference between a tcap associated with a rcv end-point, and one that is used for the scheduling decisions.  This design allows the scheduler to not have to constantly `tcap_transfer`, and spend time managing tcaps.  So, we'd like to continue to support that, which is what wakeup is all about.

What do you think?